Modules/Filtering/Smoothing/include/itkFFTDiscreteGaussianImageFilter.hxx

```diff
@@ -0,0 +1,202 @@
+#ifndef itkFFTDiscreteGaussianImageFilter_hxx
+#define itkFFTDiscreteGaussianImageFilter_hxx
+
+#include "itkFFTPadImageFilter.h"
+#include "itkForwardFFTImageFilter.h"
+#include "itkInverseFFTImageFilter.h"
+#include "itkMultiplyImageFilter.h"
+
+#include "itkGaussianImageSource.h"
+#include "itkFFTConvolutionImageFilter.h"
+
+#include "itkNeighborhoodOperatorImageFilter.h"
+#include "itkGaussianOperator.h"
+#include "itkImageRegionIterator.h"
+#include "itkProgressAccumulator.h"
+#include "itkImageAlgorithm.h"
+#include "itkMacro.h"
+
+namespace itk
+{
+
+template <typename TInputImage, typename TOutputImage>
+void
+FFTDiscreteGaussianImageFilter<TInputImage, TOutputImage>::GenerateInputRequestedRegion()
+{
+  // call the superclass' implementation of this method. this should
+  // copy the output requested region to the input requested region
+  ImageToImageFilter<TInputImage, TOutputImage>::GenerateInputRequestedRegion();
+
+  // Get pointer to input
+  typename Superclass::InputImagePointer inputPtr = const_cast<TInputImage *>(this->GetInput());
+  if (inputPtr.IsNull())
+  {
+    return;
+  }
+
+  // get a copy of the input requested region (should equal the output
+  // requested region)
+  typename TInputImage::RegionType inputRequestedRegion;
+  inputRequestedRegion = inputPtr->GetRequestedRegion();
+
+  // pad the input requested region by the operator radius
+  typename TInputImage::SizeType radius;
+  radius.Fill(0);
+  for (size_t dim = 0; dim < ImageDimension; ++dim)
+  {
+    radius[dim] = (this->GetMaximumKernelWidth() - 1) / 2;
+  }
+  inputRequestedRegion.PadByRadius(radius);
+
+  // crop the input requested region at the input's largest possible region
+  if (inputRequestedRegion.Crop(inputPtr->GetLargestPossibleRegion()))
+  {
+    inputPtr->SetRequestedRegion(inputRequestedRegion);
+    return;
+  }
+  else
+  {
+    // store what we tried to request (prior to trying to crop)
+    inputPtr->SetRequestedRegion(inputRequestedRegion);
+
+    // Couldn't crop the region (requested region is outside the largest
+    // possible region).  Throw an exception.
+    itkExceptionMacro(<< "Requested region " << inputRequestedRegion
+                      << " is (at least partially) outside the largest possible region "
+                      << inputPtr->GetLargestPossibleRegion());
+  }
+}
+
+template <typename TInputImage, typename TOutputImage>
+void
+FFTDiscreteGaussianImageFilter<TInputImage, TOutputImage>::SetMaximumError(const ArrayType arr)
+{
+  itkWarningMacro("FFTDiscreteGaussianImageFilter uses a fixed-width Gaussian kernel and ignores maximum error");
+}
+
+template <typename TInputImage, typename TOutputImage>
+void
+FFTDiscreteGaussianImageFilter<TInputImage, TOutputImage>::SetInputBoundaryCondition(
+  const InputBoundaryConditionPointerType)
+{
+  itkWarningMacro("FFTDiscreteGaussianImageFilter ignores InputBoundaryCondition, use RealBoundaryCondition instead");
```

Comment:
Same as above -- is it reasonable to ignore superclass interface elements such as `InputBoundaryCondition` if they are not needed for our implementation?

Reply:
Of course.